File: Vanilla/sources/zdoc/media/lua/shared/Library/PlayerOnGroundState.lua
---@class PlayerOnGroundState : zombie.ai.states.PlayerOnGroundState
---@field private _instance PlayerOnGroundState
PlayerOnGroundState = {}

---@public
---@param arg0 IsoGameCharacter
---@return void
function PlayerOnGroundState:execute(arg0) end

---@public
---@return PlayerOnGroundState
function PlayerOnGroundState:instance() end

---@public
---@param arg0 IsoGameCharacter
---@return void
function PlayerOnGroundState:enter(arg0) end

---@private
---@param arg0 IsoPlayer
---@return void
function PlayerOnGroundState:becomeCorpse(arg0) end

---@public
---@param arg0 IsoGameCharacter
---@return void
function PlayerOnGroundState:exit(arg0) end

File: Vanilla/sources/zomboid/zombie/ai/states/PlayerOnGroundState.java
package zombie.ai.states;

import zombie.GameTime;
import zombie.ai.State;
import zombie.characters.IsoGameCharacter;
import zombie.characters.IsoPlayer;
import zombie.network.GameClient;
import zombie.network.GameServer;

public final class PlayerOnGroundState extends State {
   private static final PlayerOnGroundState _instance = new PlayerOnGroundState();

   public static PlayerOnGroundState instance() {
      return _instance;
   }

   public void enter(IsoGameCharacter var1) {
      var1.setIgnoreMovement(true);
      ((IsoPlayer)var1).setBlockMovement(true);
      var1.setVariable("bAnimEnd", false);
   }

   public void execute(IsoGameCharacter var1) {
      if (!GameServer.bServer && var1.isDead() && var1.getVariableBoolean("bAnimEnd")) {
         this.becomeCorpse((IsoPlayer)var1);
      } else {
         var1.setReanimateTimer(var1.getReanimateTimer() - GameTime.getInstance().getMultiplier() / 1.6F);
      }

   }

   public void exit(IsoGameCharacter var1) {
      var1.setIgnoreMovement(false);
      ((IsoPlayer)var1).setBlockMovement(false);
   }

   private void becomeCorpse(IsoPlayer var1) {
      if (!var1.isOnDeathDone()) {
         if (GameClient.bClient) {
            var1.networkAI.processDeadBody();
         }

         var1.becomeCorpse();
      }

   }
}
